### DISCUSSION TITLE
Angular 16 run storybook fails after install

### DISCUSSION THREAD

**Discussion by andrew-corliss** [storybookjs/storybook#29228] [Help]
### Summary

Hello,

I am trying to integrate storybook into an existing Angular project.

However I am met with errors coming from vitest where several TS names do not seem to have definitions :
```
ERROR in node_modules/storybook/core/types/index.d.ts:2:1 - error TS1383: Only named exports may use 'export type'.

2 export type * from '@storybook/core/types';
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

ERROR in node_modules/webpack/types.d.ts:7410:5 - error TS2315: Type 'ServerOptions' is not generic.

7410   | ServerOptionsImport<typeof IncomingMessage>
         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

ERROR in node_modules/webpack/types.d.ts:13349:41 - error TS2315: Type 'ServerOptions' is not generic.

13349 > = SecureContextOptions & TlsOptions & ServerOptionsImport<Request, Response>;
                                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

```

Has anyone gotten latest version (Storybook v8) to work with Angular 16??

I know there is some setting I am missing is anyone able to help identify

**Comment by CaptainCsaba** [comment]
Hey Andrew. I might be late to the party, but the same thing happened for me, although with angular 18. I've introduced Storybook to the project I worked with. I have noticed that in the project package.json `@types/node` had a version of `"^12.12.29"`. In the storybook sourcecode the version was `"^22.0.0"`.

Updating the project `package.json` to `"@types/node": "^22.0.0"` has resolved the error for me.
